File: src/main/java/me/raatiniemi/linker/domain/LinkMap.java
package me.raatiniemi.linker.domain;

public interface LinkMap {
    String getPrefix();

    String getTarget();

    /**
     * Check if the text matches the regex.
     *
     * @param text Text to check against regex.
     * @return true if text matches, otherwise false.
     */
    boolean match(String text);
}

File: src/main/java/me/raatiniemi/linker/domain/LinkMap.kt
package me.raatiniemi.linker.domain

internal interface LinkMap {
    val prefix: String
    val target: String

    /**
     * Check if the text matches the regex.
     *
     * @param text Text to check against regex.
     * @return true if text matches, otherwise false.
     */
    fun match(text: String): Boolean
}
